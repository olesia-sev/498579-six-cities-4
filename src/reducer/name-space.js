const NameSpace = {
  DATA: `DATA`,
  APP: `APP`,
  USER: `USER`,
};

export {NameSpace};
