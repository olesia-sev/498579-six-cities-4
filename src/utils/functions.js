export const getRandomNumberInRange = (min, max) => {
  return Math.floor(Math.random() * (max - min + 1)) + min;
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};
