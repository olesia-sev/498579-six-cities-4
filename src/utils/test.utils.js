export const optionsAmount = 100;

export const testString = `Test string`;

export const emptyFunction = () => {};

export const cardDataArray = [
  {
    id: 10,
    img: `https://placedog.net/260/200?id=2`,
    images: [
      `https://placedog.net/260/200?id=3`,
      `https://placedog.net/260/200?id=33`,
      `https://placedog.net/260/200?id=12`,
      `https://placedog.net/260/200?id=10`,
      `https://placedog.net/260/200?id=11`,
      `https://placedog.net/260/200?id=15`,
    ],
    price: 150,
    title: `Title`,
    placeType: `Flat`,
    numberOfBedrooms: 2,
    maxGuests: 3,
    rating: 4,
    saved: true,
    premium: true,
    featuresInside: [
      `Wi-Fi`, `Heating`,
    ],
    hostName: `Pete`,
    hostAvatar: `https://placedog.net/260/200?id=5`,
    userPro: true,
    description: `Description`,
  },
  {
    id: 100,
    img: `https://placedog.net/260/200?id=3`,
    images: [
      `https://placedog.net/260/200?id=33`,
      `https://placedog.net/260/200?id=99`,
      `https://placedog.net/260/200?id=89`,
      `https://placedog.net/260/200?id=29`,
      `https://placedog.net/260/200?id=13`,
      `https://placedog.net/260/200?id=16`,
    ],
    price: 150,
    title: `Title test`,
    placeType: `Flat`,
    numberOfBedrooms: 1,
    maxGuests: 2,
    rating: 4,
    saved: false,
    premium: false,
    featuresInside: [
      `Wi-Fi`, `Heating`,
    ],
    hostName: `Pete`,
    hostAvatar: `https://placedog.net/260/200?id=57`,
    userPro: false,
    description: `Description`,
  }
];
