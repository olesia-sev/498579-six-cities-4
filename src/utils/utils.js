export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const SORT_TYPES = {
  POPULAR: `Popular`,
  TO_HIGH_PRICE: `Price: low to high`,
  TO_LOW_PRICE: `Price: high to low`,
  TOP_RATED: `Top rated first`,
};

export const getSortedOffers = (offers, activeSort) => {
  switch (activeSort) {
    case SORT_TYPES.POPULAR:
      return offers.slice();
    case SORT_TYPES.TO_HIGH_PRICE:
      return offers.slice().sort((a, b) => a.price - b.price);
    case SORT_TYPES.TO_LOW_PRICE:
      return offers.slice().sort((a, b) => b.price - a.price);
    case SORT_TYPES.TOP_RATED:
      return offers.slice().sort((a, b) => b.rating - a.rating);
    default:
      return offers.slice();
  }
};

export const MAX_REVIEWS_LENGTH = 10;

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  FAVOURITES: `/favorites`,
  OFFER_DETAIL: `/offers/:id`,
};

export const RATING_STAR_WIDTH = 20;

export const MAX_IMAGES_AMOUNT = 6;
export const SLICE_BEGIN = 0;
