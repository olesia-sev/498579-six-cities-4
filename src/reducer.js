import {extend} from './utils/functions';
import cities from "./mocks/cities";
import offers from "./mocks/offers";

export const initialState = {
  activeCityId: 10,
  cities,
  offers,
};

const ActionType = {
  SET_ACTIVE_CITY: `SET_ACTIVE_CITY`,
};

const ActionCreator = {
  setActiveCity: (id) => ({
    type: ActionType.SET_ACTIVE_CITY,
    payload: id,
  })
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_ACTIVE_CITY:
      return extend(state, {
        activeCityId: action.payload,
      });

    default: {
      return state;
    }
  }
};

export {reducer, ActionType, ActionCreator};
