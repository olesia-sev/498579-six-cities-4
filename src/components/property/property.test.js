import React from "react";
import renderer from "react-test-renderer";
import {MemoryRouter, Route} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import Property from "./property";
import {cardDataArray, citiesArray} from "../../utils/test.utils";
import {NameSpace} from "../../reducer/name-space";
import {SORT_TYPES} from "../../utils/functions";

const mockStore = configureStore([]);

it(`Property should be rendered`, () => {
  const store = mockStore({
    [NameSpace.APP]: {
      activeSortingType: SORT_TYPES.POPULAR,
    },
    [NameSpace.DATA]: {
      offers: cardDataArray,
      cities: citiesArray,
      activeCityId: citiesArray[0].id,
    },
  });

  const tree = renderer
    .create(
        <MemoryRouter initialEntries={[`/offers/10`]}>
          <Route path="/offers/:id">
            <Provider store={store}>
              <Property />
            </Provider>
          </Route>
        </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
