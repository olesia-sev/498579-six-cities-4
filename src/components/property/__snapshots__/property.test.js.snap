// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property should be rendered 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <span
                className="header__nav-link header__nav-link--profile"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                  style={Object {}}
                />
                <a
                  className="header__login"
                  href="/login"
                  onClick={[Function]}
                >
                  Sign in
                </a>
              </span>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Title"
              className="property__image"
              src="https://placedog.net/260/200?id=3"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Title"
              className="property__image"
              src="https://placedog.net/260/200?id=33"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Title"
              className="property__image"
              src="https://placedog.net/260/200?id=12"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Title"
              className="property__image"
              src="https://placedog.net/260/200?id=10"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Title"
              className="property__image"
              src="https://placedog.net/260/200?id=11"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Title"
              className="property__image"
              src="https://placedog.net/260/200?id=15"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Title
            </h1>
            <button
              className="property__bookmark-button button property__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Flat
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              2
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              3
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              150
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Wi-Fi
              </li>
              <li
                className="property__inside-item"
              >
                Heating
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper
                    property__avatar-wrapper--pro
                     user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="https://placedog.net/260/200?id=5"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Pete
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Description
              </p>
            </div>
          </div>
        </div>
      </div>
      <section
        className="property__map map"
      >
        <div
          id="map"
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="near-places__card place-card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="https://placedog.net/260/200?id=2"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    150
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button place-card__bookmark-button--active"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offers/10"
                  onClick={[Function]}
                >
                  Title
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Flat
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
